ocaml4.06.1-re2-0.11.0 installation fails on macOS

## Issue description

I'm unable to install `ocaml4.06.1-re2-0.11.0` package on macOS.

### Steps to reproduce

```shell
$ nix-env -i ocaml4.06.1-re2-0.11.0
```

This fails with the following error message:
```shell
installing 'ocaml4.06.1-re2-0.11.0'
these derivations will be built:
  /nix/store/hlcz14dh381yrzl4fj7lcz4pgz8brs1f-ocaml4.06.1-re2-0.11.0.drv
building '/nix/store/hlcz14dh381yrzl4fj7lcz4pgz8brs1f-ocaml4.06.1-re2-0.11.0.drv'...
unpacking sources
unpacking source archive /nix/store/xwnbi6pnzz2j1mjy5picgza3zj4drzj5-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
File "example/jbuild", line 1, characters 0-0:
Warning: jbuild files are deprecated, please convert this file to a dune file instead.
Note: You can use "dune upgrade" to convert your project to dune.
File "src/jbuild", line 1, characters 0-0:
Warning: jbuild files are deprecated, please convert this file to a dune file instead.
Note: You can use "dune upgrade" to convert your project to dune.
File "src/re2_c/jbuild", line 1, characters 0-0:
Warning: jbuild files are deprecated, please convert this file to a dune file instead.
Note: You can use "dune upgrade" to convert your project to dune.
        bash src/re2_c/dllre2_c_stubs.so,src/re2_c/libre2_c_stubs.a (exit 2)
(cd _build/default/src/re2_c && /nix/store/g9cis4hsnzcjkzfa77nif7kzq9bsqmpg-bash-4.4-p23/bin/bash -e -u -o pipefail -c '
ARFLAGS=rsc
CXX=g++
CXXFLAGS="-Wall -O3 -g -pthread"
if ! true; then
  CXX="$CXX -m32"
fi
/nix/store/y7vbmqld1dkkcl8839w98n5999s3932s-gnumake-4.2.1/bin/make -s -C libre2 clean
/nix/store/y7vbmqld1dkkcl8839w98n5999s3932s-gnumake-4.2.1/bin/make -s -C libre2 ARFLAGS="$ARFLAGS" CXX="$CXX" CXXFLAGS="$CXXFLAGS" obj/libre2.a obj/so/libre2.so
cp libre2/obj/libre2.a libre2_c_stubs.a
cp libre2/obj/so/libre2.so dllre2_c_stubs.so
/nix/store/y7vbmqld1dkkcl8839w98n5999s3932s-gnumake-4.2.1/bin/make -s -C libre2 clean
')
make: g++: Command not found
make: *** [Makefile:161: obj/util/arena.o] Error 127
        bash src/options.ml
clang-5.0: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
          cc src/stubs.o (exit 1)
(cd _build/default/src && /nix/store/1fg9nq7pr6xhbdni4f1mxxzrclbq7h26-clang-wrapper-5.0.2/bin/cc -I /nix/store/blmbclxkrgxx1f09gmbanayc6dgi9dx0-ocaml-4.06.1/lib/ocaml -I /nix/store/0dylv46lz10q1x3yblrcz74gnglz9xzj-ocaml4.06.1-ppx_bench-0.11.0/lib/ocaml/4.06.1/site-lib/ppx_bench/runtime-lib -I /nix/store/1ccqbipjzpw2kq9cqxi0qay69baprv06-ocaml4.06.1-typerep-0.11.0/lib/ocaml/4.06.1/site-lib/typerep -I /nix/store/1nzn750l4xkzvjahzy2c2k6q3mvadrbm-ocaml4.06.1-ppx_inline_test-0.11.0/lib/ocaml/4.06.1/site-lib/ppx_inline_test/config -I /nix/store/1nzn750l4xkzvjahzy2c2k6q3mvadrbm-ocaml4.06.1-ppx_inline_test-0.11.0/lib/ocaml/4.06.1/site-lib/ppx_inline_test/runtime-lib -I /nix/store/55cnmv951lp16f1bj05rkzivfz24c3dm-ocaml4.06.1-ppx_assert-0.11.0/lib/ocaml/4.06.1/site-lib/ppx_assert/runtime-lib -I /nix/store/88dsxf3plq0kii8srpg02h42zdpnkb7c-ocaml4.06.1-bin_prot-0.11.0/lib/ocaml/4.06.1/site-lib/bin_prot -I /nix/store/88dsxf3plq0kii8srpg02h42zdpnkb7c-ocaml4.06.1-bin_prot-0.11.0/lib/ocaml/4.06.1/site-lib/bin_prot/shape -I /nix/store/ccgxv74sp6shk93qnjg37ryslc713cxj-ocaml4.06.1-sexplib0-0.11.0/lib/ocaml/4.06.1/site-lib/sexplib0 -I /nix/store/h3qqpfj80vv1xk1df1v6i6r2rkmqp9zc-ocaml4.06.1-splittable_random-0.11.0/lib/ocaml/4.06.1/site-lib/splittable_random -I /nix/store/igrc16afijdr3kq4frl55q38g66j6qqy-ocaml4.06.1-ppx_sexp_conv-0.11.2/lib/ocaml/4.06.1/site-lib/ppx_sexp_conv/runtime-lib -I /nix/store/iv0wx9x70qx3d45yfg04kz2gqiczfbvv-ocaml4.06.1-ppx_compare-0.11.1/lib/ocaml/4.06.1/site-lib/ppx_compare/runtime-lib -I /nix/store/iy7w78s4v1wi7pawllwjzy9knr8q5v0p-ocaml4.06.1-stdio-0.11.0/lib/ocaml/4.06.1/site-lib/stdio -I /nix/store/j8dg4vwhlndp35l1a3sna4x9sqln33vj-ocaml4.06.1-ppx_expect-0.11.0/lib/ocaml/4.06.1/site-lib/ppx_expect/collector -I /nix/store/j8dg4vwhlndp35l1a3sna4x9sqln33vj-ocaml4.06.1-ppx_expect-0.11.0/lib/ocaml/4.06.1/site-lib/ppx_expect/common -I /nix/store/j8dg4vwhlndp35l1a3sna4x9sqln33vj-ocaml4.06.1-ppx_expect-0.11.0/lib/ocaml/4.06.1/site-lib/ppx_expect/config -I /nix/store/jjfwcnhshj66l0dlr2fppw81jyqli17m-ocaml4.06.1-fieldslib-0.11.0/lib/ocaml/4.06.1/site-lib/fieldslib -I /nix/store/ky2q66vqi0q1f79nrzwqznbwqzv4xqgg-ocaml4.06.1-ppx_hash-0.11.1/lib/ocaml/4.06.1/site-lib/ppx_hash/runtime-lib -I /nix/store/lfvfnpmcp7kfzwkaaq6vpd52v5q9j7a6-ocaml4.06.1-sexplib-0.11.0/lib/ocaml/4.06.1/site-lib/sexplib -I /nix/store/lghjsd15mjbb0wyyx9wlq4annmnlaj9l-ocaml4.06.1-parsexp-0.11.0/lib/ocaml/4.06.1/site-lib/parsexp -I /nix/store/rlx73ma2cx60dvb7ad7vjizd9j8qv134-ocaml4.06.1-variantslib-0.11.0/lib/ocaml/4.06.1/site-lib/variantslib -I /nix/store/rp3cvvvjm3g7lnfxq2743pn22ji0626d-ocaml4.06.1-jane-street-headers-0.11.0/lib/ocaml/4.06.1/site-lib/jane-street-headers -I /nix/store/y8g48bz9sfbzxk5ki7apbg414ilj7bwr-ocaml4.06.1-core_kernel-0.11.1/lib/ocaml/4.06.1/site-lib/core_kernel -I /nix/store/y8g48bz9sfbzxk5ki7apbg414ilj7bwr-ocaml4.06.1-core_kernel-0.11.1/lib/ocaml/4.06.1/site-lib/core_kernel/base_for_tests -I /nix/store/z9w6h1hf6dkmd0alvlpngsmkj91ggqz6-ocaml4.06.1-base-0.11.1/lib/ocaml/4.06.1/site-lib/base -I /nix/store/z9w6h1hf6dkmd0alvlpngsmkj91ggqz6-ocaml4.06.1-base-0.11.1/lib/ocaml/4.06.1/site-lib/base/caml -I /nix/store/z9w6h1hf6dkmd0alvlpngsmkj91ggqz6-ocaml4.06.1-base-0.11.1/lib/ocaml/4.06.1/site-lib/base/md5 -I /nix/store/z9w6h1hf6dkmd0alvlpngsmkj91ggqz6-ocaml4.06.1-base-0.11.1/lib/ocaml/4.06.1/site-lib/base/shadow_stdlib -I re2_c -O2 -fno-strict-aliasing -fwrapv -I re2_c/libre2 -g -o stubs.o -c stubs.cpp)
clang-5.0: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
stubs.cpp:1:10: fatal error: 'vector' file not found
#include <vector>
         ^~~~~~~~
1 error generated.
      ocamlc src/.re2.objs/byte/re2__Regex.{cmo,cmt}
File "src/regex.ml", line 142, characters 25-46:
Warning 52: Code should not depend on the actual values of
this constructor's arguments. They are only for information
and may change in future versions. (See manual section 8.5)
    ocamlopt src/.re2.objs/native/re2__Regex.{cmx,o}
File "src/regex.ml", line 142, characters 25-46:
Warning 52: Code should not depend on the actual values of
this constructor's arguments. They are only for information
and may change in future versions. (See manual section 8.5)
builder for '/nix/store/hlcz14dh381yrzl4fj7lcz4pgz8brs1f-ocaml4.06.1-re2-0.11.0.drv' failed with exit code 1
error: build of '/nix/store/hlcz14dh381yrzl4fj7lcz4pgz8brs1f-ocaml4.06.1-re2-0.11.0.drv' failed
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.2.0, macOS 10.14.3`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.1`
 - channels(anton): `"nixpkgs-19.09pre173147.03050e9749e"`
 - nixpkgs: `/Users/anton/.nix-defexpr/channels/nixpkgs`

--- a/pkgs/development/ocaml-modules/janestreet/default.nix
+++ b/pkgs/development/ocaml-modules/janestreet/default.nix
@@ -324,10 +324,11 @@ rec {
     pname = "re2";
     hash = "0bl65d0nmvr7k1mkkcc4aai86l5qzgn1xxwmszshpwhaz87cqghd";
     propagatedBuildInputs = [ core_kernel ];
-    meta = {
-      description = "OCaml bindings for RE2";
-      broken = stdenv.isDarwin;
-    };
+    prePatch = ''
+      substituteInPlace src/re2_c/jbuild --replace 'CXX=g++' 'CXX=c++'
+      substituteInPlace src/jbuild --replace '(cxx_flags ((:standard \ -pedantic) (-I re2_c/libre2)))' '(cxx_flags ((:standard \ -pedantic) (-I re2_c/libre2) (-x c++)))'
+    '';
+    meta.description = "OCaml bindings for RE2";
   };
 
   core_extended = janePackage {

